src/Facility.Definition/ServiceInfo.cs

```diff
@@ -155,6 +164,68 @@ ServiceMemberInfo DoExcludeTag(ServiceMemberInfo member)
 			}
 		}
 
+		private static void EnsureProperValidateUsage(ServiceTypeInfo type, ServiceFieldInfo field)
+		{
+			if (field.Validation == null) return;
+			var validateAttribute = field.TryGetAttribute("validate")!;
```

Comment:
I would expect this method to validate the `ServiceFieldValidation`, not the raw attribute.

Reply:
I implemented that at one point, but doing it this way appeared to result in better error messages since type-dependent validation happens _after_ parameter/argument validation for the attribute. e.g. if someone used `[validate(value: /\d?/)]`, the first error would be that value doesn't have a valid parameter. The second error would be there's no value parameter on the attribute because it wasn't created due to the first error.

I can look into it again, but that's why bc96563ef8a695ea04ca788d1a3765e43eb742be happened.

Reply:
I think the simpler code is worth the slightly worse error messages.